templates/usr/local/sbin/aptly_mirror_sync.sh.erb

```diff
@@ -19,3 +19,9 @@ if [[ $? -ne 0 ]]; then
 else
   /usr/bin/aptly publish switch -passphrase="<%= @gpg_pass -%>" -batch=true -component=non-free,main,contrib ${mirror_name} ${mirror_name}-nonfree-${date} ${mirror_name}-main-${date} ${mirror_name}-contrib-${date}
 fi
+
+# remove old snapshots related to this mirror
+for snapshot in $( /usr/bin/aptly snapshot list -raw=true | grep -v ${date} | egrep "^${mirror_name}-" ); do /usr/bin/aptly snapshot drop $snapshot; done
```

Comment:
Have you tried this already? I think since we are running "merge" commands on snapshots, you might get errors for snapshots that are used as base for merges already existing.